\section{MGnify}\label{sec:MGnify-intro}

MGnify offers a free-to-use service for the analysis and archiving of microbiome data.
The pipelines and used tools are open source (GitHub~\cite{noauthor_mgnify_nodate-22}). Each workflow is comprehensively documented using the Common Workflow Language (CWL)~\cite{language_cwl_home_nodate}, enhancing the reproducibility of the pipelines~\cite{mitchell_mgnify_2020}. Version 5.0 of MGnify offers three analysis pipelines associated with distinct input types: amplicon data, raw metagenomic reads, and assembly~\cite{noauthor_mgnify_nodate-25}. At the time of writing this thesis, MGnify had analyzed 356,039 amplicon analyses, 28,873 assemblies, and 33,827 metagenomic analyses~\cite{noauthor_mgnify_nodate-23}.